esmvalcore/cmor/_fixes/cmip6/fio_esm_2_0.py

```diff
@@ -0,0 +1,64 @@
+"""Fixes for FIO-ESM-2-0 model."""
+import iris
+
+from ..common import OceanFixGrid
+from ..fix import Fix
+from ..shared import round_coordinates
+
+Tos = OceanFixGrid
+
+
+class Omon(Fix):
+    """Fixes for Omon vars."""
+
+    def fix_metadata(self, cubes):
+        """Fix latitude and longitude with round to 6 d.p.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+        """
+        round_coordinates(cubes,
+                          decimals=6,
+                          coord_names=["longitude", "latitude"])
+        return cubes
+
+
+class Amon(Fix):
+    """Fixes for Amon vars."""
+
+    def fix_metadata(self, cubes):
+        """Fix metadata.
+
+        FIO-ESM-2-0 ta data contains error in coordinate bounds.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+        """
+        for cube in cubes:
+            # Check both lat and lon coords and replace bounds if necessary
+            latitude = cube.coord("latitude")
+            if latitude.has_bounds():
+                if latitude.bounds[1:, 0] != latitude.bounds[:-1, 1]:
+                    latitude.bounds = None
+                    latitude.guess_bounds()
+                    iris.util.promote_aux_coord_to_dim_coord(cube, "latitude")
+
+            longitude = cube.coord("longitude")
+            if longitude.has_bounds():
```

Comment:
I guess that if the coordinates do not have bounds they will be guessed anyway in the checks? In any case, I don't follow why you need to promote the coordinate afterwards, what is preventing it being a dimensional coordinate when the data is being loaded?

Reply:
It was so that bounds that are badly described get fixed in the specific model fix. It is true that they would be fixed anyway later on. Should I remove the bound guessing from here and expect no bounds if they are badly defined?

Regarding the promotion, it has no reason really so I'll remove it

Reply:
Nono, I was just wondering why you were only guessing the bounds in the case they were present but wrong, and doing nothing if they were not present. But since this is a fix in Amon, if some variable does not have bounds, they will be guessed anyway in the checks. So it's fine as it is.